### PR TITLE
Fix/code label

### LINKED ISSUE #200: Fix code label selector

### LINKED ISSUE #201: Keys when code appears like [Object,Object] and not as string on selectors

### PR DESCRIPTION
**Pull request by ramirobg94**
fixes #200 
fixes #201

### CODE REVIEW

File: src/arena/record/index.js [hunk 1]
```diff
@@ -1,12 +1,74 @@
...
+  nodes,
+  nodeDefsByUuid,
+  categoryItemIndex,
+}) => {
+  const keyNodes = nodes.filter(
     node =>
-      node.parentUuid === rootNode.uuid &&
+      node.parentUuid === entity.uuid &&
       nodeDefsByUuid[node.nodeDefUuid].props.key,
   );
```
Comment (SteRiccio) [inline src/arena/record/index.js]: you could call getKeyNodesForEntity instead of repeating it

File: src/arena/record/index.js [hunk 2]
```diff
@@ -1,12 +1,74 @@
...
+    nodes,
+    nodeDefsByUuid,
+  });
+
+  return getKeyNodesAsString({
+    nodes: keyRootNodes,
+    defaultString: '-',
+    joinString: '/',
+    categoryItemIndex,
+  });
```
Comment (SteRiccio) [inline src/arena/record/index.js]: what if you do just:
`getKeyNodesForEntityAsString({ entity: rootNode, ...})` and you pass some other parameter for the separator ecc?

File: src/form/Attributes/Code/Preview/hooks/useCode.js [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-    ? `(${categoryItem?.props?.code}) ${categoryItem?.props?.labels?.[language]}`
-    : '-';
+const getCategoryItemLabel =
+  nodeDef =>
+  ({categoryItem, language}) =>
+    categoryItem?.props?.code
+      ? `(${categoryItem?.props?.code}) ${
+          categoryItem?.props?.labels?.[language] || ''
+        }`
+      : '-';
```
Comment (SteRiccio) [inline src/form/Attributes/Code/Preview/hooks/useCode.js:13]: you should consider the Show code prop in the nodeDef too

File: src/form/Attributes/Code/Preview/hooks/useCode.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 import surveySelectors from 'state/survey/selectors';
 
-const getCategoryItemLabel = ({categoryItem, language}) =>
-  categoryItem?.props?.code
-    ? `(${categoryItem?.props?.code}) ${categoryItem?.props?.labels?.[language]}`
-    : '-';
+const getCategoryItemLabel =
+  nodeDef =>
+  ({categoryItem, language}) =>
+    categoryItem?.props?.code
```
Comment (SteRiccio) [inline src/form/Attributes/Code/Preview/hooks/useCode.js:9]: the code is mandatory... only categoryItem could be null

**Review by SteRiccio — APPROVED** [review]

File: src/form/Attributes/Code/Preview/hooks/useCode.js (hunk 3 above)
Comment (ramirobg94) [reply src/form/Attributes/Code/Preview/hooks/useCode.js:13]: Thanks! 
There is one PR for that, I will add them

File: src/arena/record/index.js (hunk 1 above)
Comment (ramirobg94) [reply src/arena/record/index.js]: good catch I did the refactor and I didnt notice that

**Pull request merged**